### PR TITLE
Add stronger, entropy-based password rule

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
It's time to strengthen our password rules: https://docs.google.com/document/d/1Kv_aA7vEv-uYY00rNcYu_fx3qVUxaMCeN8qJVBPj9f0/edit

#### Related Pull Requests
* https://github.com/LabKey/testAutomation/pull/1623

#### Changes
* New entropy-based password filtering/scoring mechanism and junit tests
* Update database auth dialog to show the new option
* Rename "Strong" rule to "Good"
* Admin warning if "Weak" is used on a production server
* Metrics for database auth settings

### CODE REVIEW

File: api/src/org/labkey/api/security/StrongPasswordValidator.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private final HtmlString _summaryRuleHtml;
+
+    public StrongPasswordValidator()
+    {
+        _fullRuleHtml = DOM.createHtml(DOM.createHtmlFragment(
+            DOM.UL(
+                DOM.LI("Evaluate passwords using a modern scoring approach based on entropy, a measure of the password's " +
+                    "inherent randomness. Long passwords that use multiple character types (upper case, lower case, digits, " +
+                    "symbols) result in higher scores. Repeated characters, repeated sequences, trivial sequences, and personal " +
+                    "information result in lower scores, since they are easier to guess. A visual gauge shows the strength of the " +
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/security/StrongPasswordValidator.java]: If you're confident we'll add the visual gauge soon, fine to keep it here. But could save that promise for the future PR.

**Review by labkey-jeckels — APPROVED** [review]

File: api/src/org/labkey/api/security/StrongPasswordValidator.java (hunk 1 above)
Comment (labkey-adam) [reply api/src/org/labkey/api/security/StrongPasswordValidator.java]: Fair enough... I commented out that last sentence for now

**Review by labkey-tchad — COMMENTED** [review]
TeamCity's looking happy now

**Comment by labkey-adam** [comment]
> TeamCity's looking happy now

Awesome... thanks for your help, @labkey-tchad

**Pull request merged**